### ISSUE TITLE
My Data: Does not display access via IP Groups permissions

### ISSUE THREAD

**Issue by kcondon** [IQSS/dataverse#3275]
IP Groups was recently overhauled and fixed and one of the remaining issues is My Data does not display cards where access was granted via ip groups.

**Comment by sekmiller** [comment]
@pdurbin - The key I think is to look at groupsFor method in the GroupsServiceBean. It takes in an authenticated user and returns a map of the groups the user belongs to.  The intent is to pick up run-time groups here.

**Comment by pdurbin** [comment]
I just demonstrated this bug to @raprasad and @sekmiller as of 80268c3 on the `1380-honor-ip-groups`. I'm going add them as assignees to help me figure out where in the MyData code a list of groups for the user is determined.

**Comment by sekmiller** [comment]
In the RoleAssignmentServiceBean the method getUserRuntimeGroups needs to use the new "groupsFor" from the GroupsServiceBean

**Comment by raprasad** [comment]
see: ```explicitgroup_containedroleassignees```

**Comment by pdurbin** [comment]
On Friday @raprasad and I started collaborating on a new branch called `mydata-ipgroups` which I am keeping up to date with the `1380-honor-ip-groups` branch (which pull request #3103 is based on) with the intention of merging the new branch back into the existing branch before the pull request is merged. An SQL query needs to be written in the MyData code to handle "groups within groups" such as the "IP groups within an explicit group" scenario we've been testing in #3273.

**Comment by pdurbin** [comment]
Ok @sekmiller have the "IP Group within an explicit group" test scenario set up on his machine and he's going to take a look. He can see how #3273 was fixed with the `collectAncestors` added by @michbarsinai in 1c1c60b.

**Comment by pdurbin** [comment]
This bug fix was a joint effort. @raprasad and I refactored the MyData code to allow the new `collectAncestors` method to be called and @sekmiller applied the actual fix in d50351d which I just merged into branch for the IP groups pull request #3103. That is to say that the fix is in the `1380-honor-ip-groups` branch.

The test case scenario is the same as https://github.com/IQSS/dataverse/issues/3273#issuecomment-245913997 except that after logging in, the user should still be able to see the unpublished content under "My Data".

Passing to QA.
